In [2]:
#tratamiento de datos
import pandas as pd

#configuracion de visualizacion
pd.set_option('display.max_columns',None)

#exploracion de archivos
import sys
sys.path.append('../')

#funciones personalizadas
from SRC import sp_eda as se


In [3]:
df_u = pd.read_csv('../Datos/df_data_unificada.csv')

In [4]:
se.eda_preliminar(df_u)

,Unnamed: 0_x,age,job,marital,education,default,housing,loan,contact,duration,campaign,pdays,previous,poutcome,emp.var.rate,cons.price.idx,cons.conf.idx,euribor3m,nr.employed,y,date,latitude,longitude,id_,Unnamed: 0_y,Income,Kidhome,Teenhome,Dt_Customer,NumWebVisitsMonth,ID
9742,9839,48.0,blue-collar,MARRIED,basic.9y,0.0,0.0,0.0,telephone,63,5,999,0,NONEXISTENT,1.4,"94,465","-41,8",NaN,"5228,1",no,19-julio-2019,41.932,-114.645,b401e928-fab3-4d99-9d4c-4a7f42e19716,9839,174750,0,0,2012-03-27,15,b401e928-fab3-4d99-9d4c-4a7f42e19716
24742,24864,31.0,housemaid,MARRIED,university.degree,NaN,0.0,0.0,cellular,110,1,999,0,NONEXISTENT,-0.1,"93,2",-42,"4,153","5195,8",no,15-julio-2017,47.486,-89.151,45cdc353-17e7-47ed-85c7-5d143355b697,4749,94028,2,1,2013-10-07,16,45cdc353-17e7-47ed-85c7-5d143355b697
1554,1651,31.0,blue-collar,MARRIED,basic.4y,0.0,0.0,0.0,telephone,157,1,999,0,NONEXISTENT,1.1,"93,994","-36,4","4,855",5191,no,4-febrero-2019,32.030,-74.388,3bd9bc74-fe0e-49e0-b8f2-2b7953a28e6d,1651,102319,0,0,2012-08-22,25,3bd9bc74-fe0e-49e0-b8f2-2b7953a28e6d
22990,23087,39.0,technician,MARRIED,university.degree,0.0,0.0,0.0,cellular,191,1,999,0,NONEXISTENT,1.4,"93,444","-36,1","4,965","5228,1",no,29-septiembre-2018,41.249,-102.474,72d4a2bc-36f2-4db1-be18-17f1d094b9ed,2972,150081,1,1,2013-04-13,21,72d4a2bc-36f2-4db1-be18-17f1d094b9ed
15950,16047,46.0,housemaid,DIVORCED,basic.4y,NaN,1.0,0.0,cellular,259,1,999,0,NONEXISTENT,1.4,"93,918","-42,7","4,961","5228,1",no,12-marzo-2019,28.912,-114.732,05f7742b-0a05-42e6-b02e-6c18049e285d,16047,38117,0,1,2012-03-23,19,05f7742b-0a05-42e6-b02e-6c18049e285d


------------------------
DIMENSIONES
Nuestro conjunto de datos presenta un total de 43000 filas y 31 columnas
------------------------
INFO
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 43000 entries, 0 to 42999
Data columns (total 31 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Unnamed: 0_x       43000 non-null  int64  
 1   age                37880 non-null  float64
 2   job                42655 non-null  object 
 3   marital            42915 non-null  object 
 4   education          41193 non-null  object 
 5   default            34019 non-null  float64
 6   housing            41974 non-null  float64
 7   loan               41974 non-null  float64
 8   contact            43000 non-null  object 
 9   duration           43000 non-null  int64  
 10  campaign           43000 non-null  int64  
 11  pdays              43000 non-null  int64  
 12  previous           43000 non-null  int64  
 13  poutcome           43000 n

None

------------------------
NULOS


Unnamed: 0_x          0.000000
age                  11.906977
job                   0.802326
marital               0.197674
education             4.202326
default              20.886047
housing               2.386047
loan                  2.386047
contact               0.000000
duration              0.000000
campaign              0.000000
pdays                 0.000000
previous              0.000000
poutcome              0.000000
emp.var.rate          0.000000
cons.price.idx        1.095349
cons.conf.idx         0.000000
euribor3m            21.525581
nr.employed           0.000000
y                     0.000000
date                  0.576744
latitude              0.000000
longitude             0.000000
id_                   0.000000
Unnamed: 0_y          0.000000
Income                0.000000
Kidhome               0.000000
Teenhome              0.000000
Dt_Customer           0.000000
NumWebVisitsMonth     0.000000
ID                    0.000000
dtype: float64

------------------------
DUPLICADOS
0
------------------------
FRECUENCIA CATEGORICAS
JOB
job
admin.           10873
blue-collar       9654
technician        7026
services          4162
management        3050
retired           1790
entrepreneur      1522
self-employed     1489
housemaid         1123
unemployed        1063
student            903
Name: count, dtype: int64
------------------------
MARITAL
marital
MARRIED     25999
SINGLE      12105
DIVORCED     4811
Name: count, dtype: int64
------------------------
EDUCATION
education
university.degree      12722
high.school             9925
basic.9y                6309
professional.course     5477
basic.4y                4356
basic.6y                2386
illiterate                18
Name: count, dtype: int64
------------------------
CONTACT
contact
cellular     27396
telephone    15604
Name: count, dtype: int64
------------------------
POUTCOME
poutcome
NONEXISTENT    37103
FAILURE         4461
SUCCESS         1436
Name: count, dtype: i

,count,mean,std,min,25%,50%,75%,max
Unnamed: 0_x,43000.0,20651.099047,11868.770913,0.000,10381.75000,20642.5000,30930.25000,41187.000
age,37880.0,39.977112,10.437957,17.000,32.00000,38.0000,47.00000,98.000
default,34019.0,0.000088,0.009390,0.000,0.00000,0.0000,0.00000,1.000
housing,41974.0,0.535998,0.498708,0.000,0.00000,1.0000,1.00000,1.000
loan,41974.0,0.155620,0.362499,0.000,0.00000,0.0000,0.00000,1.000
duration,43000.0,257.739279,258.666033,0.000,102.00000,179.0000,319.00000,4918.000
campaign,43000.0,2.567233,2.772294,1.000,1.00000,2.0000,3.00000,56.000
pdays,43000.0,962.330953,187.260394,0.000,999.00000,999.0000,999.00000,999.000
previous,43000.0,0.174023,0.497366,0.000,0.00000,0.0000,0.00000,7.000
emp.var.rate,43000.0,0.077128,1.573898,-3.400,-1.80000,1.1000,1.40000,1.400


## OBSERVACIONES EN LOS DATOS PRELIMINARES

* El conjunto de datos cuenta con un total de 43,000 registros y 31 columnas, lo que indica una base lo suficientemente amplia para realizar análisis estadísticos. Su estructura combina variables numéricas, categóricas y de tipo booleano (0/1).

* En cuanto a los valores faltantes, existen nulos significativos en algunas variables que podrían afectar el análisis. Las más afectadas son age (alrededor del 12%), default (21%) y euribor3m (21%), seguidas de education con un 4% y housing y loan con un 2% aproximadamente. Estas ausencias deberán ser imputadas o tratadas según su relevancia para el estudio. Por otro lado, no se detectaron registros duplicados.

* Respecto a las variables categóricas, destacan job, marital y education. Las dos primeras presentan distribuciones equilibradas, aunque los grupos “admin.”, “blue-collar” y “technician” concentran una proporción importante de los casos.

* En las variables numéricas, los rangos son amplios y existen indicios de valores atípicos, especialmente en duration, campaign y previous. Antes de cualquier modelado será necesario revisar y posiblemente estandarizar estas variables para evitar distorsiones en los resultados.

* El data set parece estar en buen estado general, pero antes del análisis será imprescindible realizar tareas de limpieza de nulos, homogeneización de categorías y tratamiento de outliers, con el objetivo de garantizar que los resultados sean consistentes y fiables.

## TRATAMIENTO DE DATOS

* Aqui se pueden observar numeras cuestiones a tratar en el siguiente jupyter de Limpieza como:
- normalizar los nombres de las columnas
- substituir espacios y puntos por guiones bajos
- convertir mayúsculas a minúsculas
- convertit floats a integers
- reemplazar guiones por guiones bajos
- eliminar el punto final de una de la categoría "admin"
- transformar "date" y "date_customer" a formato datetime
- crear nuevas columnas string a partir de booleanas.
